src/librustc/infer/error_reporting.rs

```diff
@@ -527,7 +527,7 @@ impl<'a, 'gcx, 'tcx> InferCtxt<'a, 'gcx, 'tcx> {
     {
         let expected_found = match values {
             None => None,
-            Some(values) => match self.values_str(&values) {
+            Some(ref values) => match self.values_str(&values) {
```

Comment:
Oh. I understood. We can write `self.values_str(values)` because values is reference or am I wrong?